pkg/envoy/ciliumenvoyconfig.go

```diff
@@ -760,3 +768,24 @@ func fillInTransportSocketXDS(ts *envoy_config_core.TransportSocket) {
 		}
 	}
 }
+
+// resourceQualifiedName returns the qualified name of an Envoy resource,
+// prepending CEC namespace and CEC name to the resource name and using
+// "-" as a separator.
+// In case of an empty CEC namespace or an empty CEC name, leading separators
+// are stripped away.
+func resourceQualifiedName(namespace, name, resource string) string {
+	var sb strings.Builder
+
+	if namespace != "" {
+		sb.WriteString(namespace)
+		sb.WriteRune('-')
+	}
+	if name != "" {
+		sb.WriteString(name)
+		sb.WriteRune('-')
+	}
+	sb.WriteString(resource)
+
+	return sb.String()
+}
```

Comment:
The `resourceQualifiedName` will not work in the case that one resource has the same namespace name as the other resource:
resource-1 namespace: `foo`
resource-1 name: `bar`
resource-1 resource: `hello`

resource-2 namespace: ` `
resource-2 name: `foo-bar`
resource-2 resource: `hello`

Reply:
Ouch. What about using two different separators, like `namespace/name-resource`? That should do the trick.

Reply:
the above example is kind of escaped bug we have before (e.g. in 1.12), I noticed about it a little bit later 1.12 release.

Due to the below [comment](https://github.com/cilium/cilium/blob/master/pkg/envoy/ciliumenvoyconfig.go#L412-L416), changing listener name will cause some packet drop, so we can either:
- keep the same name as 1.12 so that upgrade will work, or
- change separator convention to fix the _potential_ name overlapping bug